packages/beacon-state-transition/src/util/bellatrix.ts

```diff
@@ -10,10 +10,22 @@ import {
  * Execution enabled = merge is done.
  * When (A) state has execution data OR (B) block has execution data
  */
-export function isExecutionEnabled(state: BeaconStateBellatrix, body: bellatrix.BeaconBlockBody): boolean {
+export function isExecutionEnabled<T extends allForks.BlockType>(
+  type: T,
+  state: BeaconStateBellatrix,
+  body: allForks.FullOrBlindedBellatrixBeaconBlockBody<T>
+): boolean {
   return (
     isMergeTransitionComplete(state) ||
-    !ssz.bellatrix.ExecutionPayload.equals(body.executionPayload, ssz.bellatrix.ExecutionPayload.defaultValue())
+    (type === allForks.BlockType.Blinded
+      ? !ssz.bellatrix.ExecutionPayloadHeader.equals(
+          (body as bellatrix.BlindedBeaconBlockBody).executionPayloadHeader,
+          ssz.bellatrix.ExecutionPayloadHeader.defaultValue()
```

Comment:
This code (before this PR) is really inneficient. For each block it has to:
- Create a new data structure zero'ed from scratch
- Do a full comparison of the data structure with the block's one

Can we do better an maybe avoid having to know if the block is blinded or not? Ideas:
- Cache and compare the root of ExecutionPayloadHeader which is equivalent blinded or full
- Compare only some fields, that if they are zero, everything else must be zero (like stateRoot)

Reply:
tackled for blinded vs full by body inspection :+1: 
For efficiency issue would prefer to do a separate PR for efficiency , building on your idea: may be we can hash tree root and pass around the execution payload hash (as its anyway going to be computed/used in forkchoice if everything passes up good). similarly for cachedbeaconstate. if you like the idea, can open an issue and PR on it. :slightly_smiling_face: